src/react/useLock.ts

```diff
@@ -0,0 +1,55 @@
+import { useContext, useEffect, useState } from 'react';
+import { SpaceContext } from './contexts/SpaceContext.js';
+
+import type { LockStatus, SpaceMember, Lock } from '../types.js';
+
+interface UseLockResult {
+  status: LockStatus | null;
+  member: SpaceMember | null;
+}
+
+/*
+ * Returns the status of a lock and, if it has been acquired, the member holding the lock
+ */
+export function useLock(lockId: string): UseLockResult {
+  const space = useContext(SpaceContext);
+  const [status, setStatus] = useState<LockStatus | null>(null);
+  const [member, setMember] = useState<SpaceMember | null>(null);
+
+  const initialized = status !== null;
+
+  useEffect(() => {
+    if (!space) return;
+
+    const handler = (lock: Lock) => {
+      if (lock.id !== lockId) return;
+
+      if (lock.status === 'unlocked') {
+        setStatus(null);
+        setMember(null);
```

Comment:
FYI - this seems to cause missing rerenders - since React 16 setState(null) acts that way (but seems to be fixed again in React 18)

React 17: https://codesandbox.io/s/react-17-forked-dhx4f8
React 18: https://playcode.io/1655663

That was a fun afternoon 😬